docs/design-roadmap/ha-control-plane.md

```diff
@@ -0,0 +1,84 @@
+
+But there is no k8s-native way to achieve transactional consistency across multiple k8s clusters which each have their own etcd instances.
+
+Consequently we'll need to implement our own. We will need to decide:
+Whether to [embed](https://pkg.go.dev/github.com/coreos/etcd/embed) an etcd server in k8ssandra-operator and span it across the multiple Kubernetes clusters (which may come in handy in the future but which is a heavy weight dependency). 
+What consensus mechanism to use otherwise (Raft or a Paxos derivative are probably sensible choices).
+
+### State replication
+
+In the current design, only the control-plane cluster has a copy of the K8ssandraCluster resource. But if any k8ssandra-operator instance can be a leader, this means that all instances need to have access to the current cluster target state as represented by the K8ssandraCluster k8s object. 
+
+My first tentative suggestion here is that we will need to replicate the K8ssandraCluster object to all k8ssandra-operator instances (i.e. to all k8s clusters). 
+
+We can distinguish between those that should be acted upon (because they are in the leader's cluster) and those which are present only for resilience using an additional annotation on the resource.
+
+But an additional complication again arises in ensuring consistency. What happens if changes are made to the CR but fail to replicate before control plane failure?
+
+Our options are;
+1. Simply accept that some k8ssandra-operator instances may take over as leader with a slightly stale state and then begin reconciling towards it, in the case where a leader failure has occurred.
+2. Include replication logic in the validating webhook so that all/some minimum number of clusters must have received any updates to the K8ssandraCluster CR before it is accepted by the recipient control plane server. (Think Cassandra's write acknowledgement behavior).
+3. Investigate the use of Kubefed to see if it has functionality that might suit our use case. I would want to be certain on how it fails over in the event of whole-of-cluster loss, as there is little discussion in the Kubefed docs as to what happens in extreme failure scenarios. 
+
+Option 2 sounds attractive but may not be possible under real-world latency constraints - i.e. the HTTP connection between the client (e.g. kubectl) and the API server may timeout before the data can replicate. We would need to examine mitigations for this sort of issue.
+
+## Considerations and outstanding questions
+
+### Additional user stories that may be relevant
+
+This topic touches on a number of operational questions that users are likely to face in the real world. I'm not sure how many of these we have/want to consider, but I thought it'd be good to document them somewhere.
+
+- A user has an existing CassandraDataCenter from Cass operator. How can they join it to a new K8ssandra cluster?
```

Comment:
The procedure is documented in this blog post: https://k8ssandra.io/blog/tutorials/how-to/how-to-migrate-an-existing-cluster-to-k8ssandra-operator-without-any-downtime/

Reply:
That's good from a migration standpoint. There's a question still as to whether we support it long term.

Reply:
NB: we should also note that this ends up in a situation where the clusters are joined to the same Cassandra ring, but the cass-operator cluster isn't under the management of k8ssandra-operator. So there's two meanings to 'how can they join it to a new k8ssandra cluster' in that sentence.